tests/test_security.py

```
@@ -0,0 +1,133 @@
+    / "repo-indexer"
+    / "scripts"
+    / "git-sync.sh"
+)
+
+
+class TestCredentialRedaction:
+    """Tests for _redact_url function in git-sync.sh."""
+
+    def _run_redact(self, url: str) -> str:
+        """Run _redact_url function directly."""
+        result = subprocess.run(
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)